### ISSUE TITLE
Chapter 3: Binary Classifier

### ISSUE THREAD

**Issue by khiladi23295** [ageron/handson-ml#397]
this part of the code:

` from sklearn.linear_model import SGDClassifier
  sgd_clf = SGDClassifier(max_iter=5, tol=-np.infty, random_state=42)
  sgd_clf.fit(X_train, y_train_3)`

is giving error : The number of classes has to be greater than one; got 1 class

**Comment by khiladi23295** [comment]
i found the fix,
`y_train_3=(y_train==3) change it to y_train_3=(y_train==**'3'**)`
just had to use the inverted commas for the number.

**Comment by ageron** [comment]
Hi @khiladi23295,

When using `fetch_openml()`, the labels are returned as strings, not unsigned 8-bit integers as with other MNIST-loading methods.  So I recommend you cast them to uint8 just after loading them:

```python
from sklearn.datasets import fetch_openml
mnist = fetch_openml("mnist_784", version=1)
X, y = mnist["data"], mnist["target"]
y = y.astype(np.uint8)
 ```

This will allow you to write:

```python
y_train_3 = (y_train == 3)
```

Without any problem.
